### PR TITLE
Update space listing when map moves around

### PR DESCRIPTION
**Pull request by mathias234**
Changed the response of spaces_in_rect to 
```
{ 
  listing: "html for displaying space listings",
  markers: [array of map markers] 
}
```
And the javascript controller will then take the html generated by the server and place it under the map, I was originally planning to use TurboStream for this, but seems like this way is easier and should work just as well

### CODE REVIEW

**Comment by mathias234** [comment]
[Map: Moving the map should update the list of spaces, not just the list of markers](https://trello.com/c/IuIHxUyF/62-map-moving-the-map-should-update-the-list-of-spaces-not-just-the-list-of-markers)

File: Gemfile [hunk 1]
```diff
@@ -43,14 +43,12 @@ group :development do
   gem 'web-console', '>= 4.1.0'
   # Display performance information such as SQL time and flame graphs for each request in your browser.
   # Can be configured to work on production as well see: https://github.com/MiniProfiler/rack-mini-profiler/blob/master/README.md
-  gem 'listen', '~> 3.3'
-  gem 'rack-mini-profiler', '~> 2.0'
-  # Spring speeds up development by keeping your application running in the background. Read more: https://github.com/rails/spring
-  gem 'dotenv-rails'
```
Comment (DanielJackson-Oslo) [inline Gemfile:49]: We don't need dotenv?

**Review by DanielJackson-Oslo — CHANGES_REQUESTED** [review]
Nice! Mostly it LGTM! 

Three questions I would like to ask before merging, but not sure if we need to do anything about the first two now - or if we can fix them later in a different PR:

### Will this allow for pagination in an easy way in the future?

Will we get pagination for free if we do this in a more railsy way with turbo streams or will it be as easy with the current AJAX html-over-the-wire method?

### Can we easily only re-render the parts that change with this method? 

Would love to not see the "loading" unless there's actually new content loading, but not sure how to feasibly do that.

### Clean up linting on the js file

There's some bad linting on the js file, I fixed it in the box below. Should just be a copy/paste:

```javascript

  async loadNewMapPosition() {

    document.getElementById("space-listing").innerText = "Laster..."

    const northWest = this.map.getBounds().getNorthWest()
    const southEast = this.map.getBounds().getSouthEast()
    const fetchSpacesInRectUrl = `/spaces_in_rect?north_west_lat=${northWest.lat}&north_west_lng=${northWest.lng}&south_east_lat=${southEast.lat}&south_east_lng=${southEast.lng}`
    const spacesInRect = await (await fetch(fetchSpacesInRectUrl)).json()

    // Replace the spaces list with the new view rendered by the server
    document.getElementById("space-listing").innerHTML = spacesInRect.listing

    const markers = spacesInRect.markers

    // Remove markers that are no longer relevant
    Object.keys(this.markers).forEach(key => {
      if (markers.find( space => space.id === key)) return;
      this.removeMarker(key)
    })

```

File: Gemfile (hunk 1 above)
Comment (DanielJackson-Oslo) [reply Gemfile:49]: Doesn't seem like it! Sweet!

Comment (mathias234) [reply Gemfile:49]: It was a duplicate entry

**Comment by mathias234** [comment]
@DanielJackson-Oslo 

**RE: Pagination**
I am not sure we can use turbo streams at all, as a call can only `render` once, and we need to render the JSON for the map markers any approach I can think of will just duplicate the amount of work we have to do on both backend and frontend.
 
Neither of the approaches will give us free pagination, it will most likely have to be a parameter in the fetch URL.

**RE: Rerender**
Instead of saying `loading` we could simply just not change out the inner html content until the new data is available, although this may feel unresponsive on slow internet connections. 
As for only drawing the new spaces I think that's an unnecessary complexity, as we would have to keep track server side which spaces were added and removed (similar to how we currently do on the frontend for map markers). 
If it becomes an issue to rerender the ~10 spaces that will be shown at once we can then do that optimization which I highly doubt.

**RE: Cleaning Up**
I think there was some issue with my editor that broke space, looked fine for me. We will install ESLint to overcommit to stop this in the future

**Review by DanielJackson-Oslo — APPROVED** [review]
LGTM! Cool stuff!

**Pull request merged**